transport/config/config.go

```diff
@@ -0,0 +1,145 @@
+package config
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/allegro/akubra/metrics"
+)
+
+// ClientTransportDetail properties
+type ClientTransportDetail struct {
+	// MaxIdleConns see: https://golang.org/pkg/net/http/#Transport
+	// Default 0 (no limit)
+	MaxIdleConns int `yaml:"MaxIdleConns" validate:"min=0"`
+	// MaxIdleConnsPerHost see: https://golang.org/pkg/net/http/#Transport
+	// Default 100
+	MaxIdleConnsPerHost int `yaml:"MaxIdleConnsPerHost" validate:"min=0"`
+	// IdleConnTimeout see: https://golang.org/pkg/net/http/#Transport
+	// Default 0 (no limit)
+	IdleConnTimeout metrics.Interval `yaml:"IdleConnTimeout"`
+	// ResponseHeaderTimeout see: https://golang.org/pkg/net/http/#Transport
+	// Default 5s (no limit)
+	ResponseHeaderTimeout metrics.Interval `yaml:"ResponseHeaderTimeout"`
+	// DisableKeepAlives see: https://golang.org/pkg/net/http/#Transport
+	// Default false
+	DisableKeepAlives bool `yaml:"DisableKeepAlives"`
+}
+
+// ClientTransportTriggers properties
+type ClientTransportTriggers struct {
+	Method     string `yaml:"Method" validate:"max=64"`
+	Path       string `yaml:"Path" validate:"max=64"`
+	QueryParam string `yaml:"QueryParam" validate:"max=64"`
+}
+
+// TriggersCompiledRules properties
+type TriggersCompiledRules struct {
+	MethodRegexp     *regexp.Regexp
+	PathRegexp       *regexp.Regexp
+	QueryParamRegexp *regexp.Regexp
+	IsCompiled       bool
+}
+
+// Transport properties
+type Transport struct {
+	Name                  string                  `yaml:"Name"`
+	Triggers              ClientTransportTriggers `yaml:"Triggers"`
+	TriggersCompiledRules TriggersCompiledRules
+	Details               ClientTransportDetail `yaml:"Details"`
+}
+
+// Transports map with Transport
+type Transports []Transport
+
+// compileRule
+func (t *Transport) compileRule(regexpRule string) (compiledRule *regexp.Regexp, err error) {
+	if len(regexpRule) > 0 {
+		compiledRule, err = regexp.Compile(regexpRule)
+	}
+	return
```

Comment:
imho an error should be returned if `len(regexpRule)` < 1 - it should make sense to allow for an empty rule

Reply:
As you can see it doesn't compile empty rules. All empty rules in section `Matchers` are allow and akubra selecting `default Transport`.
If we allow empty rules in `Matchers` it's impossible to throws Error in one rule (e.g. in `else`) with current specification.